src/NServiceBus.Transport.RabbitMQ/DelayedDelivery/DelayInfrastructure.cs

```diff
@@ -78,30 +78,44 @@ public static void TearDown(IModel channel)
             }
         }
 
-        public static string CalculateRoutingKey(int delayInSeconds, string address, out int startingDelayLevel)
+        public static unsafe string CalculateRoutingKey(int delayInSeconds, string address, out int startingDelayLevel)
         {
             if (delayInSeconds < 0)
             {
                 delayInSeconds = 0;
             }
 
-            var bitArray = new BitArray(new[] { delayInSeconds });
-            var sb = new StringBuilder();
             startingDelayLevel = 0;
```

Comment:
To be extra safe, would it be better to using a local variable that we then assign before exit to startingDelayLevel? Memory on the stack doesn't need to be pinned and therefore should require the fixed keyword here but I'm not 100% certain how it works with out-variables.